app/services/web_mentions/web_mention_handler.rb

```diff
@@ -0,0 +1,37 @@
+module WebMentions
+  class WebMentionHandler
+    def initialize(canonical_url, article_url = nil)
+      @canonical_url = canonical_url
+      @article_url = article_url
+      @webmention_url = ""
+    end
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def call
+      accepts_webmention? ? send_webmention(@webmention_url) : Rails.logger.info("#{@canonical_url} doesn't support Webmentions")
+    end
+
+    def accepts_webmention?
+      document = Nokogiri::HTML.parse(URI.open(@canonical_url)).at_css("link[rel='webmention']")
+
+      if document
+        @webmention_url = document["href"]
+        return true unless Addressable::URI.parse(@webmention_url).relative?
+      end
+      false
+    rescue StandardError => e
+      Rails.logger.error("WebmentionsException: #{e}")
```

Comment:
In the event an error is raised, I would expect this method to return false. The way it is written, if we receive an error the message will return the boolean of whether or not the log was created successfully, which in most cases will be true.

Reply:
Thanks :),  i didn't know this

Reply:
Question, do we want to silently log these errors or should we be letting them get raised so we can address them?

In my opinion, given this is being executed in a Sidekiq worker in the background I would lean towards raising the error. Since it is in the background it wont disrupt any user flow and will still notify us if anything is broken. Logs are what we use to fallback on for investigations so anything put in them is not going to be seen unless something else triggers us to look. Rather than having a user report mentions not working, I would rather be notified immediately by an error.